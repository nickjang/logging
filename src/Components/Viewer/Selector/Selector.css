.range-to,
.selector-project,
.selected .form-control,
.selected {
  font-family: var(--body-font);
}

.selected {
  padding-left: 34px;
}

.selected li {
  margin-top: 0.5rem;
}

.range-to {
  margin: auto;
}

.selected .form-control {
  width: 1.25rem;
  border: 1px solid #d3d3d3;
}

.full {
  display: none;
}

.calendar-type {
  width: max-content;
  margin: auto 0.5rem auto 0;
}

.selector-project {
  width: fit-content;
}

.selector-project {
  border: 1px solid lightgray;
  margin: 2px;
  padding: 0.5rem 0.5rem;
  font-size: 1rem;
  text-align: center;
}

@media all and (min-width: 440px) {
  .selected .form-control,
  .selector-project {
    width: 5.7rem;
  }

  .short {
    display: none;
  }

  .full {
    display: block;
  }
}
