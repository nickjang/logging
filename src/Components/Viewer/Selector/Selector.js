import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CalendarPicker from '../CalendarPicker/CalendarPicker'
import SelectorContext from '../../../Context/SelectorContext';
import './Selector.css';

class Selector extends Component {
  static contextType = SelectorContext;

  handleDelete = (e) => {
    e.preventDefault();
    this.context.deleteSelector(this.props.projectId, this.props.id);
  }

  handleAddRange = (e) => {
    e.preventDefault();
    this.context.addEndRange(this.props.projectId, this.props.id);
  }

  renderEntireProject = () => {
    return (<span>Entire Project</span>);
  };

  renderDate = () => {
    return (
      <>
        <CalendarPicker
          key='calendar'
          projectId={this.props.projectId}
          selectorId={this.props.id}
          isStart={true}
          type={this.props.type}
          value={this.props.calendar.value} />
        <span>to</span>
        {
          !this.props.endRange.added
            ? <input
                type='button'
                value='+'
                aria-label={`Make the selection a range by adding an ending ${this.props.type}`}
                onClick={(e) => this.handleAddRange(e)} />
            : <CalendarPicker
                key='endRange'
                projectId={this.props.projectId}
                selectorId={this.props.id}
                isStart={false}
                type={this.props.type}
                value={this.props.endRange.value} />
        }
      </>
    );
  };

  render() {
    return (
      <li className='group-row'>
        {
          this.props.type === 'project'
            ? this.renderEntireProject()
            : this.renderDate()
        }
        <input
          type='button'
          value='x'
          aria-label='Delete this selection'
          onClick={(e) => this.handleDelete(e)} />
      </li>
    );
  };
}

Selector.defaultProps = {
  id: '',
  type: '',
  calendar: {
    value: '',
  },
  endRange: {
    added: false,
    value: '',
  },
  projectId: null
}

Selector.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['project', 'years', 'months', 'days']).isRequired,
  calendar: PropTypes.shape({
    value: PropTypes.object,
  }),
  endRange: PropTypes.shape({
    added: PropTypes.bool,
    value: PropTypes.object,
  }),
  projectId: PropTypes.number.isRequired
}

export default Selector;