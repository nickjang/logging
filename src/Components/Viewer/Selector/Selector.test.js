import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Selector from './Selector';

describe(`Selector component`, () => {
  it('renders Selector by default', () => {
    const wrapper = shallow(<Selector />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})