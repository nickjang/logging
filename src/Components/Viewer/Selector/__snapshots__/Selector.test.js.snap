// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector component renders Selector by default 1`] = `
<li
  className="group-row"
>
  <CalendarPicker
    isStart={true}
    key="calendar"
    projectId={-1}
    selectorId=""
    type="days"
    value={null}
  />
  <span
    className="range-to ml-1 mr-1"
  >
    to
  </span>
  <input
    aria-label="Make the selection a range by adding an end range."
    className="lg-btn lg-btn-light"
    onClick={[Function]}
    type="button"
    value="+"
  />
  <input
    aria-label="Delete this selection"
    className="lg-btn lg-btn-reset ml-1"
    onClick={[Function]}
    type="button"
    value="x"
  />
</li>
`;
